Source/FicsitRemoteMonitoring/Private/FRM_Library.cpp

```diff
@@ -187,4 +187,27 @@ TSharedPtr<FJsonValue> ConvertStringToFJsonValue(const FString& JsonString)
 
 	// Handle error: return nullptr or some default value
 	return nullptr;
-}
\ No newline at end of file
+}
+
+TSharedPtr<FJsonObject> UFRM_Library::getPowerConsumptionJSON(UFGPowerInfoComponent* PowerInfo) {
+	return UFRM_Library::getPowerConsumptionJSON(PowerInfo, 0);
+}
+
+TSharedPtr<FJsonObject> UFRM_Library::getPowerConsumptionJSON(UFGPowerInfoComponent* PowerInfo, float additionalPowerConsumed) {
+
+	UFGPowerCircuit* PowerCircuit = PowerInfo->GetPowerCircuit();
+	TSharedPtr<FJsonObject> JCircuit = MakeShared<FJsonObject>();
```

Comment:
I don't know the implications here for a TSharedPtr vs returning an FJsonObject. I suspect the MakeShared shouldn't be done in the helper function though.

Reply:
Honestly, I am not sure either. I probably knew at one point, but I can't find anything in my notes.

Reply:
Perusing the docs some it seems to be a reference counter that'll clean up automatically to ensure no memory leaks as C++ doesn't any garbage collection on its own. I'm leaning towards it's probably better to leave wrapped shared.

I think it's good wrapped in MakeShared like this.

Reply:
Unreal Engine, itself, has a garbage collector that I have fought against myself. So, while I'm always worried about it happening, I'm fairly confident in the current structure thus far.